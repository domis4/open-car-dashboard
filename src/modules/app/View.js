import React from 'react'
import { Provider } from 'react-redux'
import { store, history } from '../../create-store'
import { compose } from 'recompose'
import { Route, Switch } from 'react-router'
import { ConnectedRouter } from 'connected-react-router'
import { MuiThemeProvider, withStyles, createMuiTheme } from '@material-ui/core/styles'
import Fade from '@material-ui/core/Fade'
import styles from './styles'
import Bootlogo from '../bootlogo'
import Dashboard from '../dashboard'
import MainMenu from '../main-menu'
import MusicPlayer from '../music-player'
import Settings from '../settings'
import Maps from '../google-maps'
import Notifications from '../notifications'

function getTheme(theme) {
  return createMuiTheme({
    overrides: {
      MuiPaper: {
        root: {
          backgroundColor: '#000',
        },
      },
      MuiSvgIcon: {
        root: {
          color: '#fff',
        },
      },
    },
    palette: {
      type: theme.paletteType,
      background: {
        default: theme.paletteType === 'light' ? '#000' : '#fff',
      },
    },
  })
}

const theme = getTheme({
  paletteType: 'dark',
})

const App = ({ classes, isBootIconVisible, setBootIconVisible }) => (
  <Provider {...{ store }}>
    <ConnectedRouter history={history}>
      <MuiThemeProvider theme={theme}>
        <Bootlogo callback={() => setBootIconVisible(false)} />
        {!isBootIconVisible && (
          <Fade {...{ in: !isBootIconVisible, timeout: 600, className: classes.root }}>
            <div className={classes.root}>
              <div className={classes.app}>
                <MainMenu history={history} />
                <main className={classes.content}>
                  <Notifications />
                  <Switch className={classes.router}>
                    <Route exact path="/" component={Dashboard} history={history} />
                    <Route path="/music/:tab?" component={MusicPlayer} history={history} />
                    <Route path="/settings" component={Settings} history={history} />
                    <Route path="/maps" component={Maps} history={history} />
                  </Switch>
                </main>
              </div>
            </div>
          </Fade>
        )}
      </MuiThemeProvider>
    </ConnectedRouter>
  </Provider>
)

export default compose(withStyles(styles))(App)
