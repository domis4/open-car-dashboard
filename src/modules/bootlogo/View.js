import React from 'react'
import Logo from '../../assets/bootlogo/vw.svg'
import SVGInline from 'react-svg-inline'
import Fade from '@material-ui/core/Fade'
import Typography from '@material-ui/core/Typography'

const View = ({ showElement, classes, fadeIn }) => (
  <Fade {...{ in: fadeIn, timeout: 600, className: classes.root }}>
    <div {...{ className: classes.root }}>
      {showElement && (
        <div {...{ className: classes.inner }}>
          <SVGInline {...{ svg: Logo }} />
          <Typography component="h1" className={classes.text}>
            Welcome
          </Typography>
        </div>
      )}
    </div>
  </Fade>
)

export default View
