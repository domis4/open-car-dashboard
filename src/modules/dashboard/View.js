//@flow
import React from 'react'
import Paper from '@material-ui/core/Paper'
import Grid from '@material-ui/core/Grid'
import DashboardApp from '../../components/dashboard-app'
import musicImage from '../music-player/icon.png'
import settingsIcon from '../settings/icon.png'
import googleMapsIcon from '../google-maps/icon.png'

type propTypes = {
  classes: {},
  history: {},
}

const View = ({ classes, history }: propTypes) => (
  <div className={classes.root}>
    <Grid container spacing={24}>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={musicImage} title="Music" callback={() => history.push('/music')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={settingsIcon} title="Settings" callback={() => history.push('/settings')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp image={googleMapsIcon} title="Google Maps" callback={() => history.push('/maps')} />
        </Paper>
      </Grid>
      <Grid item xs={3}>
        <Paper className={classes.paper}>
          <DashboardApp />
        </Paper>
      </Grid>
    </Grid>
  </div>
)

export default View
