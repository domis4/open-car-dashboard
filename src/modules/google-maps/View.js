//@flow
import React from 'react'
import Zoom from '@material-ui/core/Zoom'

type propTypes = {
  classes: {},
  tabValue?: string,
  setTabValue: string => void,
}

const View = ({ classes, tabValue = 'lists', setTabValue }: propTypes) => (
  <Zoom in={true}>
    <iframe className={classes.iframe} src="http://maps.openrouteservice.org" scrolling="no" />
  </Zoom>
)

export default View
