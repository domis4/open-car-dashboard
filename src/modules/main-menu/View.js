import React from 'react'
import Drawer from '@material-ui/core/Drawer'
import List from '@material-ui/core/List'
import ListItem from '@material-ui/core/ListItem'
import Clock from '../../components/clock'
import ArrowBackIos from '@material-ui/icons/ArrowBackIos'
import FilterNone from '@material-ui/icons/FilterNone'

const View = ({ classes, history }) => (
  <Drawer
    variant="permanent"
    classes={{
      paper: classes.drawerPaper,
    }}
  >
    <List className={classes.top}>
      <ListItem button className={classes.center} onClick={() => history.push('/')}>
        <ArrowBackIos />
      </ListItem>
      <ListItem button className={classes.center} onClick={() => history.push('/')}>
        <FilterNone />
      </ListItem>
    </List>
    <List className={classes.bottom}>
      <ListItem className={classes.center}>
        <Clock />
      </ListItem>
      <ListItem button className={classes.center} onClick={() => history.push('/')}>
        <div className={classes.apps} />
      </ListItem>
    </List>
  </Drawer>
)

export default View
