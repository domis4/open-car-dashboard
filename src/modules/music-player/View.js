import React from 'react'
import BottomNavigation from '@material-ui/core/BottomNavigation'
import BottomNavigationAction from '@material-ui/core/BottomNavigationAction'
import ListIcon from '@material-ui/icons/List'
import FavoriteIcon from '@material-ui/icons/Favorite'
import SettingsIcon from '@material-ui/icons/Settings'
import Zoom from '@material-ui/core/Zoom'
import List from './lists'

const View = ({ classes, tabValue = 'lists', setTabValue }) => (
  <Zoom in={true}>
    <div className={classes.root}>
      <List />
      <BottomNavigation value={tabValue} onChange={(_, value) => setTabValue(value)} className={classes.bottom}>
        <BottomNavigationAction label="Lists" value="lists" icon={<ListIcon />} />
        <BottomNavigationAction label="Favorites" value="favorites" icon={<FavoriteIcon />} />
        <BottomNavigationAction label="Settings" value="settings" icon={<SettingsIcon />} />
      </BottomNavigation>
    </div>
  </Zoom>
)

export default View
