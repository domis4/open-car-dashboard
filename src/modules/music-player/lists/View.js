//@flow
import React from 'react'
import Table from '@material-ui/core/Table'
import TableBody from '@material-ui/core/TableBody'
import TableCell from '@material-ui/core/TableCell'
import TableHead from '@material-ui/core/TableHead'
import TableRow from '@material-ui/core/TableRow'
import Paper from '@material-ui/core/Paper'
import Typography from '@material-ui/core/Typography'

type propTypes = {
  classes: {},
  tabValue?: string,
  setTabValue: string => void,
}

let id = 0
function createData(name, children) {
  id += 1
  return { id, name, children }
}

const rows = [
  createData('Playlist 1'),
  createData('Playlist 2'),
  createData('Playlist 3'),
  createData('Playlist 4'),
  createData('Playlist 5'),
]

const View = ({ classes, tabValue, setTabValue }: propTypes) => (
  <Paper className={classes.root}>
    <Typography component="h1" className={classes.text}>
      Playlists
    </Typography>
    <Table className={classes.table}>
      <TableHead>
        <TableRow>
          <TableCell>Title</TableCell>
        </TableRow>
      </TableHead>
      <TableBody>
        {rows.map(row => {
          return (
            <TableRow key={row.id}>
              <TableCell component="th" scope="row">
                {row.name}
              </TableCell>
            </TableRow>
          )
        })}
      </TableBody>
    </Table>
  </Paper>
)

export default View
