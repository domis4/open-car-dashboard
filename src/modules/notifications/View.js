import React from 'react'
import Snackbar from '../../components/snackbar'

const View = ({ notifications, removeNotification }) => (
  <div>
    {notifications.map(({ uuid, text }) => (
      <Snackbar key={uuid} text={text} close={() => removeNotification(uuid)} />
    ))}
  </div>
)

export default View
