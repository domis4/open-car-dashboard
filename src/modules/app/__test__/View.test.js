import React from 'react'
import { shallow } from 'enzyme'
import View from '../View'

describe('app module View.js', () => {
  it('expects component to match snapshot', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          isBootIconVisible: false,
          setBootIconVisible: jest.fn(),
        }}
      />
    ).dive()

    expect(wrapper).toMatchSnapshot()
  })
  it('expects component contain a "Provider" component', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          isBootIconVisible: false,
          setBootIconVisible: jest.fn(),
        }}
      />
    ).dive()

    expect(wrapper.find('Provider')).toHaveLength(1)
  })
  it('expects component contain a "ConnectedRouter" component which is connected', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          isBootIconVisible: false,
          setBootIconVisible: jest.fn(),
        }}
      />
    ).dive()

    expect(wrapper.find('Connect(ConnectedRouter)')).toHaveLength(1)
  })
  it('expects component contain a "Fade" component with theme', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          isBootIconVisible: false,
          setBootIconVisible: jest.fn(),
        }}
      />
    ).dive()

    expect(wrapper.find('WithTheme(Fade)')).toHaveLength(1)
  })
  it('expects component contain a "Switch" component with theme', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          isBootIconVisible: false,
          setBootIconVisible: jest.fn(),
        }}
      />
    ).dive()

    expect(wrapper.find('Switch')).toHaveLength(1)
  })
  it('expects component contain at certain amount of "Route" components', () => {
    const wrapper = shallow(
      <View
        {...{
          classes: {},
          isBootIconVisible: false,
          setBootIconVisible: jest.fn(),
        }}
      />
    ).dive()

    expect(wrapper.find('Route')).toHaveLength(3)
  })
})
