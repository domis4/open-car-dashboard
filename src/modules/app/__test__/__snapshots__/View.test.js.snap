// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`app module View.js expects component to match snapshot 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <Connect(ConnectedRouter)
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <MuiThemeProvider
      theme={
        Object {
          "breakpoints": Object {
            "between": [Function],
            "down": [Function],
            "keys": Array [
              "xs",
              "sm",
              "md",
              "lg",
              "xl",
            ],
            "only": [Function],
            "up": [Function],
            "values": Object {
              "lg": 1280,
              "md": 960,
              "sm": 600,
              "xl": 1920,
              "xs": 0,
            },
            "width": [Function],
          },
          "direction": "ltr",
          "mixins": Object {
            "gutters": [Function],
            "toolbar": Object {
              "@media (min-width:0px) and (orientation: landscape)": Object {
                "minHeight": 48,
              },
              "@media (min-width:600px)": Object {
                "minHeight": 64,
              },
              "minHeight": 56,
            },
          },
          "overrides": Object {
            "MuiPaper": Object {
              "root": Object {
                "backgroundColor": "#000",
              },
            },
            "MuiSvgIcon": Object {
              "root": Object {
                "color": "#fff",
              },
            },
          },
          "palette": Object {
            "action": Object {
              "active": "#fff",
              "disabled": "rgba(255, 255, 255, 0.3)",
              "disabledBackground": "rgba(255, 255, 255, 0.12)",
              "hover": "rgba(255, 255, 255, 0.1)",
              "hoverOpacity": 0.1,
              "selected": "rgba(255, 255, 255, 0.2)",
            },
            "augmentColor": [Function],
            "background": Object {
              "default": "#fff",
              "paper": "#424242",
            },
            "common": Object {
              "black": "#000",
              "white": "#fff",
            },
            "contrastThreshold": 3,
            "divider": "rgba(255, 255, 255, 0.12)",
            "error": Object {
              "contrastText": "#fff",
              "dark": "#d32f2f",
              "light": "#e57373",
              "main": "#f44336",
            },
            "getContrastText": [Function],
            "grey": Object {
              "100": "#f5f5f5",
              "200": "#eeeeee",
              "300": "#e0e0e0",
              "400": "#bdbdbd",
              "50": "#fafafa",
              "500": "#9e9e9e",
              "600": "#757575",
              "700": "#616161",
              "800": "#424242",
              "900": "#212121",
              "A100": "#d5d5d5",
              "A200": "#aaaaaa",
              "A400": "#303030",
              "A700": "#616161",
            },
            "primary": Object {
              "contrastText": "#fff",
              "dark": "#303f9f",
              "light": "#7986cb",
              "main": "#3f51b5",
            },
            "secondary": Object {
              "contrastText": "#fff",
              "dark": "#c51162",
              "light": "#ff4081",
              "main": "#f50057",
            },
            "text": Object {
              "disabled": "rgba(255, 255, 255, 0.5)",
              "hint": "rgba(255, 255, 255, 0.5)",
              "icon": "rgba(255, 255, 255, 0.5)",
              "primary": "#fff",
              "secondary": "rgba(255, 255, 255, 0.7)",
            },
            "tonalOffset": 0.2,
            "type": "dark",
          },
          "props": Object {},
          "shadows": Array [
            "none",
            "0px 1px 3px 0px rgba(0, 0, 0, 0.2),0px 1px 1px 0px rgba(0, 0, 0, 0.14),0px 2px 1px -1px rgba(0, 0, 0, 0.12)",
            "0px 1px 5px 0px rgba(0, 0, 0, 0.2),0px 2px 2px 0px rgba(0, 0, 0, 0.14),0px 3px 1px -2px rgba(0, 0, 0, 0.12)",
            "0px 1px 8px 0px rgba(0, 0, 0, 0.2),0px 3px 4px 0px rgba(0, 0, 0, 0.14),0px 3px 3px -2px rgba(0, 0, 0, 0.12)",
            "0px 2px 4px -1px rgba(0, 0, 0, 0.2),0px 4px 5px 0px rgba(0, 0, 0, 0.14),0px 1px 10px 0px rgba(0, 0, 0, 0.12)",
            "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 5px 8px 0px rgba(0, 0, 0, 0.14),0px 1px 14px 0px rgba(0, 0, 0, 0.12)",
            "0px 3px 5px -1px rgba(0, 0, 0, 0.2),0px 6px 10px 0px rgba(0, 0, 0, 0.14),0px 1px 18px 0px rgba(0, 0, 0, 0.12)",
            "0px 4px 5px -2px rgba(0, 0, 0, 0.2),0px 7px 10px 1px rgba(0, 0, 0, 0.14),0px 2px 16px 1px rgba(0, 0, 0, 0.12)",
            "0px 5px 5px -3px rgba(0, 0, 0, 0.2),0px 8px 10px 1px rgba(0, 0, 0, 0.14),0px 3px 14px 2px rgba(0, 0, 0, 0.12)",
            "0px 5px 6px -3px rgba(0, 0, 0, 0.2),0px 9px 12px 1px rgba(0, 0, 0, 0.14),0px 3px 16px 2px rgba(0, 0, 0, 0.12)",
            "0px 6px 6px -3px rgba(0, 0, 0, 0.2),0px 10px 14px 1px rgba(0, 0, 0, 0.14),0px 4px 18px 3px rgba(0, 0, 0, 0.12)",
            "0px 6px 7px -4px rgba(0, 0, 0, 0.2),0px 11px 15px 1px rgba(0, 0, 0, 0.14),0px 4px 20px 3px rgba(0, 0, 0, 0.12)",
            "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 12px 17px 2px rgba(0, 0, 0, 0.14),0px 5px 22px 4px rgba(0, 0, 0, 0.12)",
            "0px 7px 8px -4px rgba(0, 0, 0, 0.2),0px 13px 19px 2px rgba(0, 0, 0, 0.14),0px 5px 24px 4px rgba(0, 0, 0, 0.12)",
            "0px 7px 9px -4px rgba(0, 0, 0, 0.2),0px 14px 21px 2px rgba(0, 0, 0, 0.14),0px 5px 26px 4px rgba(0, 0, 0, 0.12)",
            "0px 8px 9px -5px rgba(0, 0, 0, 0.2),0px 15px 22px 2px rgba(0, 0, 0, 0.14),0px 6px 28px 5px rgba(0, 0, 0, 0.12)",
            "0px 8px 10px -5px rgba(0, 0, 0, 0.2),0px 16px 24px 2px rgba(0, 0, 0, 0.14),0px 6px 30px 5px rgba(0, 0, 0, 0.12)",
            "0px 8px 11px -5px rgba(0, 0, 0, 0.2),0px 17px 26px 2px rgba(0, 0, 0, 0.14),0px 6px 32px 5px rgba(0, 0, 0, 0.12)",
            "0px 9px 11px -5px rgba(0, 0, 0, 0.2),0px 18px 28px 2px rgba(0, 0, 0, 0.14),0px 7px 34px 6px rgba(0, 0, 0, 0.12)",
            "0px 9px 12px -6px rgba(0, 0, 0, 0.2),0px 19px 29px 2px rgba(0, 0, 0, 0.14),0px 7px 36px 6px rgba(0, 0, 0, 0.12)",
            "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 20px 31px 3px rgba(0, 0, 0, 0.14),0px 8px 38px 7px rgba(0, 0, 0, 0.12)",
            "0px 10px 13px -6px rgba(0, 0, 0, 0.2),0px 21px 33px 3px rgba(0, 0, 0, 0.14),0px 8px 40px 7px rgba(0, 0, 0, 0.12)",
            "0px 10px 14px -6px rgba(0, 0, 0, 0.2),0px 22px 35px 3px rgba(0, 0, 0, 0.14),0px 8px 42px 7px rgba(0, 0, 0, 0.12)",
            "0px 11px 14px -7px rgba(0, 0, 0, 0.2),0px 23px 36px 3px rgba(0, 0, 0, 0.14),0px 9px 44px 8px rgba(0, 0, 0, 0.12)",
            "0px 11px 15px -7px rgba(0, 0, 0, 0.2),0px 24px 38px 3px rgba(0, 0, 0, 0.14),0px 9px 46px 8px rgba(0, 0, 0, 0.12)",
          ],
          "shape": Object {
            "borderRadius": 4,
          },
          "spacing": Object {
            "unit": 8,
          },
          "transitions": Object {
            "create": [Function],
            "duration": Object {
              "complex": 375,
              "enteringScreen": 225,
              "leavingScreen": 195,
              "short": 250,
              "shorter": 200,
              "shortest": 150,
              "standard": 300,
            },
            "easing": Object {
              "easeIn": "cubic-bezier(0.4, 0, 1, 1)",
              "easeInOut": "cubic-bezier(0.4, 0, 0.2, 1)",
              "easeOut": "cubic-bezier(0.0, 0, 0.2, 1)",
              "sharp": "cubic-bezier(0.4, 0, 0.6, 1)",
            },
            "getAutoHeightDuration": [Function],
          },
          "typography": Object {
            "body1": Object {
              "color": "#fff",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "0.875rem",
              "fontWeight": 400,
              "lineHeight": "1.46429em",
            },
            "body2": Object {
              "color": "#fff",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "lineHeight": "1.71429em",
            },
            "button": Object {
              "color": "#fff",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "0.875rem",
              "fontWeight": 500,
              "textTransform": "uppercase",
            },
            "caption": Object {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "0.75rem",
              "fontWeight": 400,
              "lineHeight": "1.375em",
            },
            "display1": Object {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "2.125rem",
              "fontWeight": 400,
              "lineHeight": "1.20588em",
            },
            "display2": Object {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "2.8125rem",
              "fontWeight": 400,
              "lineHeight": "1.13333em",
              "marginLeft": "-.02em",
            },
            "display3": Object {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "3.5rem",
              "fontWeight": 400,
              "letterSpacing": "-.02em",
              "lineHeight": "1.30357em",
              "marginLeft": "-.02em",
            },
            "display4": Object {
              "color": "rgba(255, 255, 255, 0.7)",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "7rem",
              "fontWeight": 300,
              "letterSpacing": "-.04em",
              "lineHeight": "1.14286em",
              "marginLeft": "-.04em",
            },
            "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
            "fontSize": 14,
            "fontWeightLight": 300,
            "fontWeightMedium": 500,
            "fontWeightRegular": 400,
            "headline": Object {
              "color": "#fff",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "1.5rem",
              "fontWeight": 400,
              "lineHeight": "1.35417em",
            },
            "pxToRem": [Function],
            "round": [Function],
            "subheading": Object {
              "color": "#fff",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "1rem",
              "fontWeight": 400,
              "lineHeight": "1.5em",
            },
            "title": Object {
              "color": "#fff",
              "fontFamily": "\\"Roboto\\", \\"Helvetica\\", \\"Arial\\", sans-serif",
              "fontSize": "1.3125rem",
              "fontWeight": 500,
              "lineHeight": "1.16667em",
            },
          },
          "zIndex": Object {
            "appBar": 1100,
            "drawer": 1200,
            "mobileStepper": 1000,
            "modal": 1300,
            "snackbar": 1400,
            "tooltip": 1500,
          },
        }
      }
    >
      <withState(withState(lifecycle(WithStyles(View))))
        callback={[Function]}
      />
      <WithTheme(Fade)
        className="App-root-01"
        in={true}
        timeout={600}
      >
        <div
          className="App-root-01"
        >
          <div
            className="App-app-02"
          >
            <WithStyles(View)
              history={
                Object {
                  "action": "POP",
                  "block": [Function],
                  "createHref": [Function],
                  "go": [Function],
                  "goBack": [Function],
                  "goForward": [Function],
                  "length": 1,
                  "listen": [Function],
                  "location": Object {
                    "hash": "",
                    "pathname": "/",
                    "search": "",
                    "state": undefined,
                  },
                  "push": [Function],
                  "replace": [Function],
                }
              }
            />
            <main
              className="App-content-03"
            >
              <Switch
                className="App-router-04"
              >
                <Route
                  component={[Function]}
                  exact={true}
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  path="/"
                />
                <Route
                  component={[Function]}
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  path="/music/:tab?"
                />
                <Route
                  component={[Function]}
                  history={
                    Object {
                      "action": "POP",
                      "block": [Function],
                      "createHref": [Function],
                      "go": [Function],
                      "goBack": [Function],
                      "goForward": [Function],
                      "length": 1,
                      "listen": [Function],
                      "location": Object {
                        "hash": "",
                        "pathname": "/",
                        "search": "",
                        "state": undefined,
                      },
                      "push": [Function],
                      "replace": [Function],
                    }
                  }
                  path="/settings"
                  when="always"
                />
              </Switch>
            </main>
          </div>
        </div>
      </WithTheme(Fade)>
    </MuiThemeProvider>
  </Connect(ConnectedRouter)>
</Provider>
`;
